common/io/io.cpp

```diff
@@ -245,7 +245,7 @@ ExpectedOfstream OpenOfstream(const string &path, bool append) {
 			generic_category().default_error_condition(io_errno),
 			"Failed to open '" + path + "' for writing")));
 	}
-	return ExpectedOfstream(std::move(os));
+	return os;
```

Comment:
This is the same, the original code is just more explicit which I like better. But YMMV.

Reply:
I don't like the original because it makes it look more like an error return. They are visually easier to distinguish when they look like this, I think:

```c++
return expected::unexpected(err);
return os;

# as opposed to

return expected::unexpected(err);
return ExpectedOfstream(std::move(os));
```

I think it also expresses intent better, since `Expected` is kind of an implementation detail in this context. What you really want is `os`.

That being said, I am glad that `expected::unexpected` is required in error context, because of the type ambiguity it can lead to otherwise.

Reply:
```
return expected::unexpected(err);
return ExpectedOfstream(std::move(os));
```
These two look different enough for me, but like I said YMMV. And this is not at all important. Both things use RVO and result in the same assembly I believe/hope.